app/routes.py

```diff
@@ -1 +1,217 @@
+        
+    return jsonify(task_response)
+
+@task_bp.route("/<task_id>", methods=["GET"])
+def read_one_task(task_id):
+    task = validate_model(Task, task_id)
+    return {"task": task.to_dict()}
+
+@task_bp.route("/<task_id>", methods=["PUT"])
+def update_task(task_id):
+    task = validate_model(Task, task_id)
+
+    request_body = request.get_json()
+
+    task.title = request_body["title"]
+    task.description = request_body["description"]
+
+    db.session.commit()
+
+    return make_response({"task": task.to_dict()})
+ 
+def slack_bot_message(message):
+    PATH = "https://slack.com/api/chat.postMessage"
+    SLACK_API_KEY = os.environ.get("SLACK_API_KEY")
+
+    query_params = {
+        "channel" : "#task-notification",
+        "text" : message
+    }
+
+    requests.post(PATH, params=query_params,headers={"Authorization": SLACK_API_KEY})
+
+@task_bp.route("/<task_id>/mark_complete", methods=["PATCH"])
+def complete_incomplete_task(task_id):
+    task = validate_model(Task, task_id)
+
+    task.completed_at = datetime.utcnow()
+    
+    db.session.commit()
+
+    slack_bot_message(f"Someone just completed the task {task.title}")
+
+    return make_response({"task": task.to_dict()})
+
+@task_bp.route("/<task_id>/mark_incomplete", methods=["PATCH"])
+def mark_complete_task_incomplete(task_id):
+    task = validate_model(Task, task_id)
+
+    task.completed_at = None
+
+    db.session.commit()
+
+    return make_response({"task": task.to_dict()})
+        
+
+@task_bp.route("/<task_id>", methods=["DELETE"])
+def delete_task(task_id):
+    task = validate_model(Task, task_id)
+
+    db.session.delete(task)
+    db.session.commit()
+
+    return make_response({"details":f'Task {task_id} "{task.title}" successfully deleted'})
+
+################## GOAL ROUTES ######################
+
+
+@goal_bp.route("", methods=["POST"])
+def handle_goals():
+    request_body = request.get_json()
+    try:
+        new_goal = Goal(title=request_body["title"])
+    except:
+        abort(make_response({"details": "Invalid data"}, 400))
+
+    db.session.add(new_goal)
+    db.session.commit()
+
+    return make_response({"goal":{
+            "id": new_goal.goal_id,
+            "title": new_goal.title
+        }}, 201)
+
+@goal_bp.route("", methods=["GET"])
+def read_all_goals():
+
+    goal = Goal.query.all()
+    goal_response = []
+    for goal in goal:
+        goal_response.append({
+            "id": goal.goal_id,
+            "title": goal.title
+        })
+
+    return jsonify(goal_response)
+
+@goal_bp.route("/<goal_id>", methods=["GET"])
+def read_one_goal(goal_id):
+    goal = validate_model(Goal, goal_id)
+    return {"goal":{
+            "id": goal.goal_id,
+            "title": goal.title,
+        }}
+
+@goal_bp.route("/<goal_id>", methods=["PUT"])
+def update_goal(goal_id):
+    goal = validate_model(Goal, goal_id)
+
+    request_body = request.get_json()
+
+    goal.title = request_body["title"]
```

Comment:
We should have error handling here since we're using data from the client to make a change to the DB. It can make your code more robust to assume that there could be errors in the request body and handle them accordingly.